arch/risc-v/src/common/riscv_cpuidlestack.c

```diff
@@ -37,8 +37,8 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
-#define SMP_STACK_MASK       15
-#define SMP_STACK_SIZE       (CONFIG_IDLETHREAD_STACKSIZE & ~15)
+#ifdef CONFIG_SMP
```

Comment:
What about non-SMP case?

Reply:
`up_cpu_idlestack()` will only be called in SMP case. If `CONFIG_SMP_NCPUS == 1`, there will be the following warning:
```
common/riscv_cpuidlestack.c: In function 'up_cpu_idlestack':
common/riscv_cpuidlestack.c:114:28: warning: array subscript <unknown> is outside array bounds of 'uint8_t[0][800]' {aka 'unsigned char[][800]'} [-Warray-bounds=]
  114 |   stack_alloc = (uintptr_t)g_cpux_idlestack[cpu - 1];
      |                            ^~~~~~~~~~~~~~~~
common/riscv_cpuidlestack.c:52:3: note: while referencing 'g_cpux_idlestack'
   52 |   g_cpux_idlestack[CONFIG_SMP_NCPUS - 1][SMP_STACK_SIZE];
```
I will narrow the scope of CONFIG_SMP to ensure that it can be compiled in non-SMP cases.